components/input/Input.js

```diff
@@ -17,9 +15,9 @@ import theme from './theme.css';
 
 export default class Input extends PureComponent {
   static getDerivedStateFromProps(nextProps, prevState) {
-    const value = nextProps.input ? nextProps.input.value : nextProps.value;
-    if (value !== undefined) {
-      const newValue = Input.parseValue(Input.getPropsValue(nextProps), nextProps);
+    if (nextProps.value !== undefined) {
```

Comment:
Is it on purpose that null is a valid value and undefined is not?

Reply:
Yes it's on purpose. It could be that the parent container wants to set the value to null.

Reply:
what this check is doing is checking wether a value has been supplied by the parent component (i.e. if this is not the case the value of `value` will be `undefined`).

If a parent decides to for whatever reason pass `null` then it means the parent has passed a value and this component shouldn't care about it anymore.

If a value is indeed `undefined` though, the Input component will AFAIK keep track of it in its state etc.

Reply:
Thanks for the explanation :)